haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellRewriter.java

```diff
@@ -380,7 +380,12 @@ private RewriterResult executeKoreCommands(
       public RewriterResult prove(Module rules, Boolean reuseDef) {
         Module kompiledModule = KoreBackend.getKompiledModule(module, true);
         ModuleToKORE converter =
-            new ModuleToKORE(kompiledModule, def.topCellInitializer, kompileOptions, kem);
+            new ModuleToKORE(
+                kompiledModule,
+                def.topCellInitializer,
+                kompileOptions,
+                // this enables ModuleToKore to print functional claims
+                kProveOptions.allowFuncClaims ? null : kem);
```

Comment:
I don't think that setting the exception manager for the entire `ModuleToKORE` class to `null` is the right way to implement this; it might require a bit more code + a duplicated constructor but I'd rather have an `allowFuncClaims` specific parameter here.